Tests/RxSwift/AccumulatingBufferTests.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  BufferTests.swift
+//  RxSwiftExt-iOS
+//
+//  Created by Patrick Maltagliati on 11/12/18.
+//  Copyright © 2018 RxSwiftCommunity. All rights reserved.
+//
+
+import XCTest
+
+import RxSwift
+import RxSwiftExt
+import RxTest
+
+class AccumulatingBufferTests: XCTestCase {
+    let testError = NSError(domain: "dummyError", code: -232, userInfo: nil)
+    let scheduler = TestScheduler(initialClock: 0)
+
+    func testBuffersUntilBoundaryEmits() {
+        let underlying = scheduler.createHotObservable(
+            [
+                next(150, 1),
+                next(201, 2),
+                next(230, 3),
+                next(300, 4),
+                next(350, 5),
+                next(375, 6),
+                next(400, 7),
+                next(430, 8),
+                completed(500)
+            ]
+        )
+
+        let boundary = scheduler.createHotObservable(
+            [
+                next(201, ()),
+                next(301, ()),
+                next(401, ())
+            ]
+        )
+
+        let res = scheduler.start(disposed: 1000) {
+            underlying.accumulatingBuffer(boundary.asObservable())
+        }
+
+        XCTAssertEqual(res.events, [
+            next(201, [2]),
+            next(301, [3, 4]),
+            next(401, [5, 6, 7]),
+            next(500, [8]),
+            completed(500)
```

Comment:
My question here is - is a `completed` really considered a boundary emission? that kinda seems like an unwanted side-effect. Basically a developer could do `take(1)` so I guess it's fine, but was that the desired behavior?

Reply:
Yes, this gives the observer of this new operator one last chance to receive buffered events.